libirods_rule_engine_plugin-elasticsearch.cpp

```diff
@@ -400,16 +383,12 @@ namespace
 	                                  const std::string& _index_name)
 	{
 		try {
-			std::string doc_type{"text"}; // TODO What is this for?
-			apply_document_type_policy(_rei, _object_path, _source_resource, &doc_type);
```

Comment:
Why are we removing document type?

Reply:
This is specific to mapping types. Mapping types were deprecated in ESv7 and removed completely in ESv8.

@trel and I had a small conversation about this and decided it was unnecessary since `doc_type` isn't used anymore.

Thinking about this more, we could keep this and include it as a new property when indexing a data object. I believe that actually aligns with the original intent.

Thoughts?

Reply:
You're suggesting we keep it, but not because of using it in the URL, just rather... to have it?

I don't know that this is the place for that.   Seems tools like https://tika.apache.org/ can do that outside of this codebase.

Reply:
Including the type information in the json document is an alternative approach to the now removed mapping types feature.

I'm not necessarily saying we have to keep it. I'm just pointing out it's possible we can continue to support it. I'm also assuming that is its intended usage. I could be wrong.

Reply:
Okay, just wanted to make sure we weren't losing some critical functionality here. Sounds like you have thought about it and have deemed it safe for removal as it is either no longer relevant or is handled elsewhere?

Reply:
While it's safe to remove, I think it needs to be investigated and discussed a little more before moving forward.

Reply:
Sounds good

Reply:
After some discussion, we've decided to move forward with the removal of this code.